map-layer-controls.js

```diff
@@ -1006,128 +1033,210 @@ class MapLayerControl {
         if (isChecked) {
             sourceControl.classList.remove('collapsed');
 
-            if (group.type === 'vector') {
-                const layerId = `vector-layer-${group.id}`;
-                if (this._map.getLayer(layerId)) {
-                    this._map.setLayoutProperty(layerId, 'visibility', 'visible');
-                    this._map.setLayoutProperty(`${layerId}-outline`, 'visibility', 'visible');
+            if (group.type === 'terrain') {
+                // Enable terrain with default settings
+                this._map.setTerrain({ 
+                    'source': 'mapbox-dem',
+                    'exaggeration': 1.5 
+                });
+                
+                // Get existing fog settings or use defaults
+                const existingFog = this._map.getFog() || {
+                    'range': [-1, 2],
+                    'horizon-blend': 0.3,
+                    'color': '#ffffff',
+                    'high-color': '#add8e6',
+                    'space-color': '#d8f2ff',
+                    'star-intensity': 0.0
+                };
+
+                // Set fog using existing or default values
+                this._map.setFog(existingFog);
+
+                // Update the UI controls to match existing values
+                const $fogStartSlider = $(sourceControl).find('.fog-range-slider input').first();
+                const $fogEndSlider = $(sourceControl).find('.fog-range-slider input').last();
+                const $horizonSlider = $(sourceControl).find('.mt-4 input[type="range"]').first();
+                const $colorPicker = $(sourceControl).find('input[type="color"]').eq(0);
+                const $highColorPicker = $(sourceControl).find('input[type="color"]').eq(1);
+                const $spaceColorPicker = $(sourceControl).find('input[type="color"]').eq(2);
```

Comment:
nit: we don't need to change this now. fro future – might be good to give ids to these input elements so we can explicitly `.find(` then.

Reply:
Good call, added ids to to the elements